pkg/virt-controller/watch/vmi.go

```diff
@@ -1282,9 +1300,22 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 		}
 
 		if vmiSpecIfaces, vmiSpecNets, dynamicIfacesExist := network.CalculateInterfacesAndNetworksForMultusAnnotationUpdate(vmi); dynamicIfacesExist {
-			if err := c.updateMultusAnnotation(vmi.Namespace, vmiSpecIfaces, vmiSpecNets, pod); err != nil {
+			errMessage := fmt.Errorf("failed to hot{un}plug network interfaces for vmi [%s/%s]", vmi.GetNamespace(), vmi.GetName())
+
+			var networkToIPAMClaimParams map[string]network.IPAMClaimParams
```

Comment:
The PR description says hotplug is TBD. This code is invoked on hotplug only, so I"m a bit confused.

Reply:
Even on non hotplug,
we will enter this code in the beginning because the status isn't updated yet and differs from the desired one
(`CalculateInterfacesAndNetworksForMultusAnnotationUpdate`).
So we need to have the info here, otherwise the new annotations will miss the new key, the pod annotations will be updated, removing the needed key, which is wrong.